src/Invitation.ts

```diff
@@ -130,20 +159,18 @@ export class SealedInvitationHeaderV2
     return this._inboxKeyBundle
   }
 
-  public async getPeerAddress(selfAddress: string): Promise<string> {
-    if (this._peerAddress) return this._peerAddress
+  public getPeerAddress(selfAddress: string): string {
     const senderAddress = this.sendKeyBundle.getLinkedAddress(
```

Comment:
@jazzz reviewed my libxmtp code and mentioned that the invitation could be between two random parties (maliciously forwarded). It'll get caught in DH derivation -> decryption failure but can also bubble it up here as "self is not part of invitation".